app/models/enterprise_role.rb

```diff
@@ -6,4 +6,24 @@ class EnterpriseRole < ActiveRecord::Base
   validates_uniqueness_of :enterprise_id, scope: :user_id, message: I18n.t(:enterprise_role_uniqueness_error)
 
   scope :by_user_email, joins(:user).order('spree_users.email ASC')
+
+  def self.receives_notifications_for(enterprise_id)
+    manager = EnterpriseRole.find_by_enterprise_id_and_receives_notifications(enterprise_id, true)
+    return nil if manager.blank?
+    Spree::User.find(manager.user_id)
+  end
+
+  def self.set_notification_user(user_id, enterprise_id)
+    managers_for(enterprise_id).map do |m|
+      if m.user_id == user_id.to_i
+        m.update_attributes receives_notifications: true
+      elsif m.user_id != user_id.to_i && m.receives_notifications
```

Comment:
the `!=` is useless here. At this point we know `m.user_id == user_id.to_i` is not true. Otherwise the `update_attributes` would be executed.